docs/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,446 @@
+
+TODO: This should be standardized in the future for consistency.
+
+Example:
+
+```
+src/
+  components/
+    SearchForm/
+      SearchForm.tsx
+      __tests__/
+        SearchForm.test.tsx
+  utils/
+    searchUtils.ts
+    utilsTests/
+      searchUtils.test.ts
+```
+
+### Components
+
+- You should be able to tell whether or not a child component will render from the parent
+- Child components should not make a determination based on a prop and return null
+- This improves readability and makes the component hierarchy more predictable
+
+Example:
+
+```typescript
+// Avoid
+const ChildComponent = ({ shouldRender, data }) => {
+  if (!shouldRender) return null
+  return <div>{data}</div>
+}
+
+const ParentComponent = ({ data }) => {
+  return (
+    <div>
+      <ChildComponent shouldRender={!!data} data={data} />
+    </div>
+  )
+}
+
+// Prefer
+const ChildComponent = ({ data }) => {
+  return <div>{data}</div>
+}
+
+const ParentComponent = ({ data }) => {
+  return <div>{data && <ChildComponent data={data} />}</div>
+}
+```
+
+### Code Formatting
+
+The project uses ESLint and Prettier for code formatting:
+
+```bash
+# Check code style
+npm run lint
+
+# Format code
+npm run prettier
+```
+
+These checks are automatically run as pre-commit hooks using Husky.
+
+## Testing Guidelines
+
+### Test Coverage
+
+All new code should include appropriate tests:
+
+- Unit tests for utility functions and small components
+- Integration tests for complex components and pages
+
+NB: While we currently do not have a coverage threshold in place (though we may in the future), the current rule of thumb is to add coverage for any practical scenario a user may experience, including edge cases when appropriate.
+
+### Running Tests
+
+```bash
+# Run all tests
+npm test
+
+# Run tests with coverage
+npm run coverage
+
+# Run tests in watch mode
+npm run test-watch
+```
+
+### Test Structure
+
+Follow these guidelines for test structure:
+
+- Group related tests in `describe` blocks
+- Use clear and descriptive test names
+- Mock external dependencies
+- Test edge cases and error conditions, especially when they cause unexpected behaviors and bugs.
+
+Example:
+
+```typescript
+describe("SearchForm", () => {
+  it("submits the search query when the form is submitted", async () => {
+    // Arrange
+    const handleSearch = jest.fn()
+    render(<SearchForm onSearch={handleSearch} />)
+
+    // Act
+    await userEvent.type(screen.getByLabelText("Search"), "test query")
+    await userEvent.click(screen.getByRole("button", { name: "Search" }))
+
+    // Assert
+    expect(handleSearch).toHaveBeenCalledWith({
+      query: "test query",
+      field: "all",
+    })
+  })
+})
+```
+
+## Pull Request Process
+
+1. **Create a Pull Request (PR)** from your feature branch to the `development` branch
```

Comment:
Use `main` instead of `development` (I also don't think there is a development branch).